docs/modeling/bounding-boxes.rst

```diff
@@ -0,0 +1,201 @@
+:math:`10^{-7}`.
+
+        >>> from astropy.modeling import models
+        >>> import numpy as np
+
+        >>> g1 = models.Gaussian1D(stddev=3)
+        >>> g1.amplitude = (g1.stddev * np.sqrt(2 * np.pi)) ** -1
+        >>> g1.bounding_box
+        (-16.5, 16.5)
+
+        >>> x1, x2 = g1.bounding_box
+        >>> x = np.linspace(x1, x2, 100)
+        >>> flux = np.trapz(g1(x), x)
+        >>> print 'Total Flux Error = %.2e' % (1 - flux)
+        Total Flux Error = 3.92e-08
+
+The limits can be set manually by passing an array-like object of 
+shape ``(model.n_inputs, 2)`` or ``(2 * model.n_inputs)``. 
+
+        >>> g2 = models.Gaussian2D(x_stddev=1, y_stddev=2)
+        >>> g2.bounding_box
+        ((-11.0, 11.0), (-5.5, 5.5))
+
+        >>> g2.bounding_box = [-4.5, 4.5, -3.0, 3.0]
+        >>> g2.bounding_box
+        ((-4.5, 4.5), (-3.0, 3.0))
+
+The default is reset with ``'auto'``.
+
+        >>> g2.bounding_box = 'auto'
+        >>> g2.bounding_box
+        ((-11.0, 11.0), (-5.5, 5.5))
+
+The ``bounding_box`` is turned with `None`.
+
+        >>> g2.bounding_box = None
+        >>> g2.bounding_box
+        >>>
+
+The default box for Gaussian models is :math:`\pm 5.5 \sigma` from the mean, 
+but this factor can be passed to ``bounding_box_default`` for the ``Gaussian1D`` 
+and ``Gaussian2D`` models.
+
+        >>> g1.stddev = 3
+        >>> g1.bounding_box = g1.bounding_box_default(factor=10)
+        >>> g1.bounding_box
+        (-30.0, 30.0)
+
+The default function can also be set to any callable. This is particularly 
+useful for fitting ``custom_model`` or ``CompoundModel`` instances. 
+
+        >>> g1.bounding_box = 'auto'
+        >>> original_default = g1.bounding_box_default
+        >>> def new_default(power=2):
+        ...    dx = g1.stddev**power
+        ...    return (g1.mean - dx, g1.mean + dx)
+        ...
+        >>> g1.bounding_box_default = new_default
+        >>> g1.bounding_box
+        (-9.0, 9.0)
+
+Efficient evaluation with ``render_model``
+------------------------------------------
+
+When a model is evaluated over a range much larger than the model itself, it may 
+be prudent to use `astropy.modeling.render_model` if efficiency is a concern. 
+The ``render_model`` function can be used to evaluate a model on an array of the 
+same dimensions. If no array is given, ``render_model`` will return a "postage 
+stamp" array corresponding to the bounding box region. However, if 
+``bounding_box`` is `None` an image or coordinates must be passed. To update the 
+input array in place, set ``copy = False``.
+
+    .. plot::
+        :include-source:
+
+        import numpy as np
+        import matplotlib.pyplot as plt
+        from astropy.modeling import models, render_model
+
+        imshape = (450, 600)
+        nsrc = 50
+        np.random.seed(0)
+
+        # Generate a random list of models:
+
+        model_params = [dict(x_mean=np.random.rand() * imshape[1],
```

Comment:
Could you use `np.random.uniform()` here (for all parameters)? It think  it looks much cleaner...and it"s easier to see what the boundaries of the parameters are.